internal/jobs/launch_instance_aws.go

```diff
@@ -144,27 +144,37 @@ func DoEnsurePubkeyOnAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) e
 	fingerprint := pubkey.FindAwsFingerprint(ctx)
 	ec2Name, err := ec2Client.GetPubkeyName(ctx, fingerprint)
 	if err != nil {
-		span.SetStatus(codes.Error, "key error")
+		logger.Trace().Msgf("Pubkey with fingerprint '%s' not found, importing type %s key '%s'", fingerprint, pubkey.Type, pubkey.Body)
 
 		// if not found on AWS, import
 		if errors.Is(err, http.PubkeyNotFoundErr) {
 			pkr.Tag = ""
 			pkr.RandomizeTag()
+			retryCount := 0
+			origName := pubkey.Name
+		retry:
 			pkr.Handle, err = ec2Client.ImportPubkey(ctx, pubkey, pkr.FormattedTag())
 
 			if errors.Is(err, http.DuplicatePubkeyErr) {
+				retryCount += retryCount + 1
```

Comment:
This would be a sequence of: 0, 1, 3, 7, etc. Is that intentional?
```suggestion
				retryCount += 1
```

Reply:
Haha made my day, no obviously this was a typo.